src/rpc/handlers/BookChanges.cpp

```diff
@@ -0,0 +1,250 @@
+#include <ripple/app/ledger/Ledger.h>
+#include <ripple/basics/ToString.h>
+
+#include <backend/BackendInterface.h>
+#include <rpc/RPCHelpers.h>
+
+#include <boost/json.hpp>
+#include <algorithm>
+
+namespace json = boost::json;
+using namespace ripple;
+
+namespace RPC {
+
+struct BookChange
+{
+    STAmount sideAVolume;
+    STAmount sideBVolume;
+    STAmount highRate;
+    STAmount lowRate;
+    STAmount openRate;
+    STAmount closeRate;
+};
+
+class BookChangesHandler
+{
+    std::reference_wrapper<Context const> context_;
+    std::map<std::string, BookChange> tally_;
+    std::optional<uint32_t> offerCancel_;
+
+public:
+    ~BookChangesHandler() = default;
+    explicit BookChangesHandler(Context const& context)
+        : context_{std::cref(context)}
+    {
+    }
+
+    BookChangesHandler(BookChangesHandler const&) = delete;
+    BookChangesHandler(BookChangesHandler&&) = delete;
+    BookChangesHandler&
+    operator=(BookChangesHandler const&) = delete;
+    BookChangesHandler&
+    operator=(BookChangesHandler&&) = delete;
+
+    /**
+     * @brief Handles the `book_change` request for given transactions
+     *
+     * @param transactions The transactions to compute changes for
+     * @return std::vector<BookChange> The changes
+     */
+    std::vector<BookChange>
+    handle(LedgerInfo const& ledger)
+    {
+        reset();
+
+        for (auto const transactions =
+                 context_.get().backend->fetchAllTransactionsInLedger(
+                     ledger.seq, context_.get().yield);
+             auto const& tx : transactions)
+        {
+            handleBookChange(tx);
+        }
+
+        std::vector<BookChange> changes;
+        std::transform(
+            std::make_move_iterator(std::begin(tally_)),
+            std::make_move_iterator(std::end(tally_)),
+            std::back_inserter(changes),
+            [](auto obj) { return obj.second; });
+        return changes;
+    }
+
+private:
+    inline void
+    reset() noexcept
+    {
+        tally_.clear();
```

Comment:
Why do we need `reset()` .

If the class is instantiated each time we handle, couldn't we just put this functionality in the constructor.

Reply:
Just because in my code i don't reuse it does not mean one could not reuse it some day later. So this is for correctness sake and it also does nothing on the first call to it as far as i understand